frontends/concrete-python/concrete/fhe/compilation/server.py

```diff
@@ -415,9 +432,20 @@ def run(
                 message = f"Expected argument {i} to be an fhe.Value but it's None"
                 raise ValueError(message)
 
-            if not isinstance(arg, Value):
-                message = f"Expected argument {i} to be an fhe.Value but it's {type(arg).__name__}"
-                raise ValueError(message)
+            elif not isinstance(arg, Value):
+                if i not in self._clear_input_indices:
+                    message = (
+                        f"Expected argument {i} to be an fhe.Value "
+                        f"but it's {friendly_type_format(type(arg))}"
+                    )
+                    raise ValueError(message)
+
+                if self._client_to_prepare_clear_values is None:
+                    self._client_to_prepare_clear_values = Client(self.client_specs)
+                    self._client_to_prepare_clear_values.keygen()
```

Comment:
The keygen should not occurs, I known we may change things a bit deeper to allow that.

Reply:
I've used simulated value exporter as the underlying implementation seems to bu just what we wanted!